server/log_rpc_server.go

```diff
@@ -324,9 +319,16 @@ func (t *TrillianLogRPCServer) GetInclusionProofByHash(ctx context.Context, req
 	if err := tx.Commit(); err != nil {
 		return nil, err
 	}
+	if len(proofs) < 1 {
+		return nil, status.Errorf(codes.NotFound,
+			"No leaf found for hash: %x in tree size %v", req.LeafHash, req.TreeSize)
+	}
 
-	r.Proof = proofs
-	return r, nil
+	// TODO(gbelvin): Rename "Proof" -> "Proofs"
```

Comment:
I think `Proof` is correct -- there's a single proof of inclusion for a single leaf in the API response.  

(What's a bit confusing is that this function only deals with a single leaf, but is written to deal with multiple leaves; i.e. the local variable `proofs` has an overly general name).

Reply:
`Proof` is also an a repeated proof object.  Is that intentional?

Reply:
Ah, I missed the [helpful comment in the API](https://github.com/google/trillian/blob/master/trillian_log_api.proto#L202-L205) that one leaf could get multiple inclusion proofs if the underlying tree allows dups (which I don't think is possible anymore?).

It does have a `TODO(gbelvin) only return one proof.` attached to it which kind of contradicts the TODO you're adding here though .... :-)

Reply:
Hmm, sounds like `TODO(gbelvin) only return one proof` should be removed?